iroha_p2p/src/network.rs

```diff
@@ -30,6 +31,8 @@ where
 {
     /// Current peers in any state
     pub peers: HashMap<PeerId, Addr<Peer<T, K, E>>>,
+    /// A set of connected peers, ready to work with
+    pub connected_peers: HashSet<PeerId>,
```

Comment:
Do you need this hashset here? You have hashmap with peers which can be used by what

Reply:
Then we need to wrap `PeerId`s in some other structure with Id and status. I've chosen to have separate hashset for that.

Reply:
Didnt get the usecase still though. You can use `HashMap::keys` for the same case

Reply:
That hashmap contains all possible ids. We need to keep statuses of them somewhere.

Reply:
We need to be connected to everyone as i know, so dont see a reason to keep all addresses if they dont hold connection.

Reply:
They maybe in a process of connection, or reconnection. Only on successful handshake peers send `Connected` message to `Network`

Reply:
I see. Generally i dont think that peers should be accessable during handshake. Like having peers which are doing handshake is kinda useless. May be peers can be spawned and as they got through handshake they can notify network with address of peer itself.

Reply:
Maybe you're right.

Reply:
Okay, we can't make it like this, as we've got to store `Addr` objects of peers ahead of time.
Or we have a possibility to get `Addr` of actor that sent us some message?